services/cd-service/pkg/repository/repository.go

```diff
@@ -1182,9 +1186,15 @@ func (r *repository) updateArgoCdApps(ctx context.Context, state *State, env str
 				TeamName: team,
 			})
 		}
-		if manifests, err := argocd.Render(r.config.URL, r.config.Branch, config, env, appData); err != nil {
+		span.Finish()
+
+		span, ctx = tracer.StartSpanFromContext(ctx, "RenderAndWrite")
```

Comment:
This make `RenderAndWrite` a subspan of `collectData`

Reply:
fixed!